src/index.js

```diff
@@ -1,21 +1,122 @@
 'use strict';
 
 const amqplib = require('amqplib');
-const { ValidationError } = require('moleculer');
+const {
+  ValidationError,
+  ServiceSchemaError,
+} = require('moleculer');
+
+const SERVICE_STATUSES = {
+  CREATED: 'CREATED',
+  STARTED: 'STARTED',
+  STOPPED: 'STOPPED',
+};
 
 /**
  * AMQP mixin
  *
  * Provide work tasks queues for rabbitMQ
  *
  * @param {String} url RabbitMQ connection string
- * @param {Object} options Connection socket options
+ * @param {Object} socketOptions Connection socket options
  * @returns {*}
  */
-module.exports = function createService(url, options) {
+module.exports = function createService(url, socketOptions) {
   return {
     name: 'amqp',
+    metadata: {
+      reconnectTimeout: 5000,
+    },
     methods: {
+      /**
+       * Connect to rabbitMQ with socketOptions.
+       * Reconnect is supported
+       *
+       * @returns {Promise<void>}
+       * @private
+       */
+      async _connect() {
+        let connectionTimeout;
+        const reconnect = () => {
```

Comment:
Maybe extract this to methods?

Reply:
Done, check pls https://github.com/Ashhm/moleculer-amqp/pull/7/commits/319de97836594a87319cbc59729ad8861f072141